sound/soc/sof/ipc4-topology.c

```diff
@@ -815,6 +817,74 @@ static void sof_ipc4_widget_free_comp_mixer(struct snd_sof_widget *swidget)
 	swidget->private = NULL;
 }
 
+/*
+ * Add the process modules support. The process modules can be EQFIR, EQIIR,
+ * KEYWORD_DETECT, KPB (KeyPhrase Buffer Manager), CHAN_SELECTOR, SMART_AMP
+ * and etc.
+ */
+static int sof_ipc4_widget_setup_comp_process(struct snd_sof_widget *swidget)
+{
+	struct snd_soc_component *scomp = swidget->scomp;
+	struct sof_ipc4_process *process;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+	struct snd_sof_control *scontrol;
+	struct sof_ipc4_control_data *control_data;
+	void *cfg;
+	int cfg_size;
+	int ret;
+
+	process = kzalloc(sizeof(*process), GFP_KERNEL);
+	if (!process)
+		return -ENOMEM;
+
+	swidget->private = process;
+
+	ret = sof_ipc4_get_audio_fmt(scomp, swidget, &process->available_fmt, true);
+	if (ret)
+		goto err;
+
+	cfg_size = sizeof(struct sof_ipc4_base_module_cfg) +
+		swidget->output_cnt * sizeof(struct sof_ipc4_audio_format);
+	/* allocate the cfg for process modules */
+	list_for_each_entry(scontrol, &sdev->kcontrol_list, list)
+		if (scontrol->comp_id == swidget->comp_id) {
+			if (sof_ipc4_ctl_is_initinstance(scontrol)) {
```

Comment:
I would suggest to rename sof_ipc4_ctl_is_initinstance to `sof_ipc4_ctl_for_instance_init` or `sof_ipc4_ctl_for_inst_init`

Reply:
OK, will use "sof_ipc4_ctl_for_inst_init"